<!-- Launch file for the rrt-detector and the assigner -->


<launch>

  <include file="$(find mbot_navigation)/launch/gmapping.launch"/>

    <!-- 运行move_base节点 -->
   <include file="$(find mbot_navigation)/launch/move_base.launch" />

<arg name="eta" value="1.0"/>
<arg name="Geta" value="15.0"/>
<param name="namespace_init_count" value="1"/>

  <node pkg="mbot_explore" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="/map"/>
  </node>
  
  <node pkg="mbot_explore" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="/map"/>
  <param name="robot_frame" value="base_footprint"/>
  </node>
 
  
  <node pkg="mbot_explore" type="filter.py" name="filter" output="screen">
  <param name="map_topic" value="/map"/>
  <param name="info_radius" value="1"/> 
  <param name="costmap_clearing_threshold" value="70"/> 
  <param name="goals_topic" value="/detected_points"/>
  <param name="namespace" value=""/> 
  <param name="n_robots" value="1"/>
  <param name="rate" value="100"/>
  <param name="robot_frame" value="base_footprint"/>
  </node>
  
  <node pkg="mbot_explore" type="assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="/map"/>
  <param name="global_frame" value="map"/>
  <param name="info_radius" value="1"/> 
  <param name="info_multiplier" value="3.0"/> 
  <param name="hysteresis_radius" value="3.0"/> 
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="n_robots" value="1"/>
  <param name="namespace" value=""/>
  <param name="delay_after_assignement" value="0.5"/>
  <param name="rate" value="100"/>
  <param name="robot_frame" value="base_footprint"/>
  </node>
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mbot_explore)/rviz/rrt.rviz">
  </node>


</launch>
